keylime/cmd/convert_ima_policy.py

```diff
@@ -109,12 +110,14 @@ def _convert_flat_format_allowlist(alist_raw):
 
 
 # Updates an existing IMA policy to the latest version, and adds any provided input
-def update_ima_policy(policy, excludelist_path=None, verification_keys=None):
+def update_ima_policy(
+    policy: dict, excludelist_path: Optional[str] = None, verification_keys: Optional[list[str]] = None
+) -> dict:
     if policy["meta"]["version"] < ima.IMA_POLICY_CURRENT_VERSION:
         print(
             f"Provided policy has version {policy['meta']['version']}; latest policy has version {ima.IMA_POLICY_CURRENT_VERSION}. Updating to latest version."
         )
-        updated_policy = copy.deepcopy(ima.IMA_POLICY_CURRENT_VERSION)
+        updated_policy: dict = copy.deepcopy(policy)
```

Comment:
IIUC it should be `ima.IMA_POLICY_CURRENT_VERSION`.  Seems to be that the idea is to iterate over the expected keys (from the default template), and build from there.

Reply:
It looked wrong to me and pyright  says:
```
/home/stefanb/keylime/keylime/cmd/convert_ima_policy.py
  /home/stefanb/keylime/keylime/cmd/convert_ima_policy.py:120:46 - error: Argument of type "Literal[1]" cannot be assigned to parameter "x" of type "_T@deepcopy" in function "deepcopy"
    Type "int" cannot be assigned to type "dict[Unknown, Unknown]"
      "int" is incompatible with "dict[Unknown, Unknown]" (reportGeneralTypeIssues)
```

Reply:
I see, maybe was `ima.EMPTY_IMA_POLICY` at some time. I am pretty sure that it should not be `policy`, that is the object that we are transforming. 

cc: @mbestavros

Reply:
If you look at line 130 there is where the copying happens, so I think @aplanas is right. I think that line 130 this should be a deep copy and we need to make sure that we return a fully copy of the policy and do not modify the old once accidentally.

Reply:
Line 130 looks good to me. I agree though that the above then should be `copy.deepcopy(ima.EMPTY_IMA_POLICY)`

Reply:
If `policy[key]` is a dict we only copy the reference. I don't know if that is intentional.

Reply:
@aplanas @stefanberger @THS-on That line was definitely a mistake - it's supposed to deepcopy `ima.EMPTY_IMA_POLICY`. The `IMA_POLICY_CURRENT_VERSION` field is just a version number, hence the type error.